### PR TITLE
chore: bump alloy to fix #8830

### LINKED ISSUE #8830: bug(`anvil`): Anvil returns transaction log format lacking expected fields for `ots_searchTransactionsBefore`, possibly others
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (8a51b89 2024-09-07T10:53:35.909768350Z)

### What command(s) is the bug in?

_No response_

### Operating System

None

### Describe the bug

For `ots_searchTransactionsBefore` and possibly other Otterscan RPC methods which return receipts, Anvil is returning a `logs` object that does not contain all the required fields per the execution API spec:

Example logs returned by Anvil (`ots_searchTransactionsBefore` response -> `receipts`[0] -> `logs`):
```json
[
	{
		"address": "0x242c735479f3b5cd6a907a90d640edcc1eb1c815",
		"data": "0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000254c6f7420230000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
		"topics": [
			"0xd9d4de58fddbbf888ad9e901d6b06fbc30f8267115058fb6860dda919f01bc79"
		]
	}
]
```

Expected format (example):
```json
[
	{
		"address": "0x248e21b0aa161efe3045e3d067d972cd6a01d1b5",
		"topics": [
			"0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
			"0x000000000000000000000000f39fd6e51aad88f6f4ce6ab8827279cfffb92266",
			"0x000000000000000000000000d7f387d4b0eece0a099d35895152de42bc987c5b",
			"0x00000000000000000000000000000000000000000000000000000000000019a7"
		],
		"data": "0x",
		"blockNumber": "0x13a6cb6",
		"transactionHash": "0xea6829000055ded8be1e45e390f0c8edab33aea6362da834e099644bef8dab75",
		"transactionIndex": "0x38",
		"blockHash": "0xca1480043b81baa375c8dfbfb4219f427a03ed126407006ee7f753389042360d",
		"logIndex": "0xc1",
		"removed": false
	}
]
```

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Fixes #8830 

## Solution

Bumps alloy to include https://github.com/alloy-rs/alloy/pull/1261

Consequently bump `op-alloy-rpc-types`

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
this fix is now released

**Comment by yash-atreya** [comment]
> this fix is now released

Patch removed

File: Cargo.toml [hunk 1]
```diff
@@ -266,29 +266,3 @@ tower-http = "0.5"
 soldeer = "=0.3.4"
 proptest = "1"
 comfy-table = "7"
-
-[patch.crates-io]
-# https://github.com/alloy-rs/alloy/pull/1229 + https://github.com/alloy-rs/alloy/pull/1243
```
Comment (mattsse) [inline Cargo.toml:271]: lets keep them as comments because we'll need them again eventually

**Review by mattsse — CHANGES_REQUESTED** [review]
nits

**Review by mattsse — APPROVED** [review]

**Pull request merged**